platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/HttpClientBuilder.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.solr.factory.impl;
+
+import ddf.security.encryption.EncryptionService;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivilegedAction;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLContext;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContexts;
+import org.apache.http.impl.auth.BasicScheme;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.solr.client.solrj.impl.PreemptiveAuth;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpClientBuilder implements Supplier<org.apache.http.impl.client.HttpClientBuilder> {
+
+  private static final String KEY_STORE_PASS = "javax.net.ssl.keyStorePassword";
+  private static final String TRUST_STORE = "javax.net.ssl.trustStore";
+  private static final String TRUST_STORE_PASS = "javax.net.ssl.trustStorePassword";
+  private static final String KEY_STORE = "javax.net.ssl.keyStore";
+  private static final String HTTPS_PROTOCOLS = "https.protocols";
+  private static final String HTTPS_CIPHER_SUITES = "https.cipherSuites";
+  private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientBuilder.class);
+  private static final String SOLR_HTTP_URL = "solr.http.url";
+  private final EncryptionService encryptionService;
+
+  public HttpClientBuilder(EncryptionService encryptionService) {
+    this.encryptionService = encryptionService;
+  }
+
+  public org.apache.http.impl.client.HttpClientBuilder get() {
+
+    org.apache.http.impl.client.HttpClientBuilder httpClientBuilder =
```

Comment:
✏️ Are we using the fully qualified path for the same reason as here?

https://github.com/codice/ddf/pull/4160#discussion_r243377607

Reply:
yes